crates/bevy_ui/src/focus.rs

```diff
@@ -143,13 +143,19 @@ pub struct NodeQuery {
     target_camera: Option<&'static TargetCamera>,
 }
 
+/// A component that can be attached to a camera to pass mouse coordinates through to ui textures, allowing
+/// manual control of interactivity for world-space uis, or custom cursor position control for window-based uis.
+/// For optimal latency the contents should be updated in [`bevy_app::PreUpdate`], before [`crate::UiSystem::Focus`] and after [`bevy_input::InputSystem`].
+#[derive(Component, Default)]
+pub struct ManualCursorPosition(pub Option<Vec2>);
```

Comment:
I don't really love adding more complexity to the focus system, especially if we are in the middle up upstreaming mod_picking, which can handle this case more generally.

Reply:
i think the complexity is minimal, and i suspect gpu picking would be unable to return the uv coords of an intersection which is required for determining the cursor position. it also seems like it would be inefficient for me when i have collision infrastructure already.

Reply:
For the mod_picking stuff being upstreamed, there are a few ways to tackle this, but they are agnostic to what backend you are using to do your hit tests. 

The input mapping is separate from the actual hit testing. You could chain these together (hit test against UI quads, pipe that as input into the UI picking backend), or you could take the simpler approach (without a frame of lag) that this PR takes, which is to just compute the pointer position on any world space UI quads in the input system, and let the picking backends consume that.

Either way, `ManualCursorPosition` won't be needed.